packages/core/src/common/command.ts

```diff
@@ -269,10 +287,12 @@ export class CommandRegistry implements CommandService {
      */
     // tslint:disable-next-line:no-any
     async executeCommand<T>(commandId: string, ...args: any[]): Promise<T | undefined> {
-        await this.fireWillExecuteCommand(commandId);
+        const executionId = Date.now().toString();
+        await this.fireWillExecuteCommand(commandId, executionId, args);
```

Comment:
On the fast machines we have today, can we rely on no two commands executing within a millisecond of each other?  Should we encode the command and args in this string too?  Alternatively, a number field incremented by one each time would work and be simpler.